CriteoPublisherSdk/Sources/CR_BidManager.h

```diff
@@ -71,6 +71,24 @@
 
 @end
 
+@interface CR_BidManager (Bidding)
+
+/**
+ * Get a bid for an Ad Unit.
+ *
+ * Depending on `liveBidding` being enabled through configuration, the bid will be either:
+ * - Fetched asynchronously using the live bidding strategy (time budget / cache fallback)
+ * - Returned synchronously using the cache bidding strategy (prefetch / get from cache)
+ *
+ * @param adUnit The ad unit to request a bid
+ * @param responseHandler The block called once bid has been loaded
+ * Note: responseHandler is potentially not invoked on main queue
```

Comment:
The action following loading a bid/ad is to be done on the main thread (displaying the Ad), it would be better if publishers don't have to do any cumbersome logic to figure out which thread the callback is being invoked on.

What prevents us from scheduling on I/O thread and observing on main thread, therefore always invoking `responseHandler` on the main thread?

Reply:
I agree publishers should not have to handle this.
This method is private and main queue logic is handled from callers only when needed, as it is not always the case, this is just a reminder to us :)
If we expose a similar method from public `Criteo`, then there will be handled the main queue.